src/System.Runtime/tests/System/Text/StringBuilderTests.netcoreapp.cs

```diff
@@ -199,5 +199,84 @@ public static void Insert_CharSpan_Invalid()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("index", () => builder.Insert(builder.Length + 1, new ReadOnlySpan<char>(new char[0]))); // Index > builder.Length
             AssertExtensions.Throws<ArgumentOutOfRangeException>("requiredLength", () => builder.Insert(builder.Length, new ReadOnlySpan<char>(new char[1]))); // New length > builder.MaxCapacity
         }
+
+        private static IEnumerable<object[]> Append_StringBuilder_TestData()
+        {
+            string mediumString = new string('a', 30);
+            string largeString = new string('b', 1000);
+
+            var sb1 = new StringBuilder("Hello");
+            var sb2 = new StringBuilder("one");
+            var sb3 = new StringBuilder(20).Append(mediumString);
+
+            yield return new object[] { new StringBuilder("Hello"), sb1, "HelloHello" };
+            yield return new object[] { new StringBuilder("Hello"), sb2, "Helloone" };
+            yield return new object[] { new StringBuilder("Hello"), new StringBuilder(), "Hello" };
```

Comment:
Is empty SB interesting case? If yes, then adding reversed order case (`{ new, "Hello", "Hello" }`) and `{ new, new, "" }` might be interesting.

Reply:
Yeah it will be interesting when first SB is empty